server/container_create.go

```diff
@@ -818,11 +819,31 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr container.Conta
 	idMapSupport := s.ContainerServer.Runtime().RuntimeSupportsIDMap(sb.RuntimeHandler())
 	rroSupport := s.ContainerServer.Runtime().RuntimeSupportsRROMounts(sb.RuntimeHandler())
 
-	containerVolumes, ociMounts, err := s.addOCIBindMounts(ctx, ctr, mountLabel, s.config.RuntimeConfig.BindMountPrefix, s.config.AbsentMountSourcesToReject, maybeRelabel, skipRelabel, cgroup2RW, idMapSupport, rroSupport, s.ContainerServer.Config().Root)
+	var cleanupSafeMounts []*safeMountInfo
+
+	runtime.LockOSThread()
+
+	cleanupFunc := func() {
+		runtime.UnlockOSThread()
+
+		for _, s := range cleanupSafeMounts {
+			s.Close()
+		}
+	}
+
+	defer func() {
+		if err != nil {
+			cleanupFunc()
+		}
+	}()
```

Comment:
Hm, we may have to do this in scope of `CreateContainer()` and not `createSandboxContainer()`. :thinking: